### DISCUSSION TITLE
Required characteristics of feature views in the on-demand feature

### DISCUSSION THREAD

**Discussion by ndenStanford** [feast-dev/feast#4372] [Q&A] [answered]
I have some questions related to the on-demand feature view. Any recommendation or suggestion is much appreciated. 

Regarding the script from the [on-demand feature view](https://docs.feast.dev/v/v0.36-branch/reference/alpha-on-demand-feature-view), what do driver_hourly_stats and transformed_conv_rate need to have in common? Do they need to share the same entity or entity join key, or do they need to have some common columns?

```
training_df = store.get_historical_features(
    entity_df=entity_df,
    features=[
        "driver_hourly_stats:conv_rate",
        "driver_hourly_stats:acc_rate",
        "driver_hourly_stats:avg_daily_trips",
        "transformed_conv_rate:conv_rate_plus_val1",
        "transformed_conv_rate:conv_rate_plus_val2",
    ],
).to_df()
```

Is there a way to find a complete example script of the on-demand feature view?

What's the rationale for needing to register an additional feature view? Initially, I thought the on-demand feature would be appended to the original in place.

Some more context 
https://github.com/feast-dev/feast/issues/4363

**Comment by tokoko** [comment, marked as answer]
They don't really need to have anything in common, odfvs don't have a concept of an entity. your `entity_df` will first be used to retrieve data from concrete feature views (so `entity_df` must contain whatever join keys `driver_hourly_stats` requires) and then transformation inside odfv will be applied to the retrieved dataset to generate new features.

The main rationale for having a separate feature view for odfvs is that they could depend on more than one unrelated feature views. Those feature views might even have different entities. odfvs don't really care about that as they are basically row-level transformations applied after all other feature views have already been retrieved.

Have you had a look at [quickstart](https://github.com/feast-dev/feast/blob/master/examples/quickstart/quickstart.ipynb) notebook?

**Reply by ndenStanford** [reply]
Thanks a lot, @tokoko. I will test this and let you know.

**Reply by ndenStanford** [reply]
@tokoko Hello Tornike. Hope you are well today. I'm still encountering the issue with the ambiguous column error in Redshift. When using SQL query in an entity_df. 

```
feast.errors.RedshiftQueryError: Redshift SQL Query failed to finish. Details: {'ClusterIdentifier': 'redshift-dev', 'CreatedAt': datetime.datetime(2024, 8, 1, 17, 57, 55, 312000, tzinfo=tzlocal()), 'Database': 'sources_dev', 'DbUser': 'admin', 'Duration': -1, 'Error': 'ERROR: column reference conv_rate is ambiguous'
```

Is this the right way to structure the entity_df as SQL query in the on-demand feature context? 

```
entity_df="""SELECT t1.conv_rate, t1.acc_rate, t1.avg_daily_trips, entity_key, event_timestamp FROM "feast"."driver_hourly_stats" t1
    LIMIT 10""" 
```
Thank you in advance.

**Comment by ndenStanford** [comment]
Sorted on my end. The query was not set up appropriately.

**Comment by ndenStanford** [comment]
Thank you @tokoko
